drivers/sqlcommons.go

```diff
@@ -0,0 +1,98 @@
+package drivers
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"strings"
+)
+
+func multiDescribeHelper(tables []string, processedTables map[string]bool, db *sql.DB, d Driver) (map[string][]FieldDescriptor, []string, error) {
+	knownTables := make(map[string]bool)
+	tableDescriptorMap := make(map[string][]FieldDescriptor)
+	var newlyReferencedTables []string
+	for _, table := range tables {
+		knownTables[table] = true
+	}
+	for _, table := range tables {
+		fields, err := d.Describe(table, db)
+		if err != nil {
+			return nil, nil, err
+		}
+		for _, field := range fields {
+			if field.ForeignKeyDescriptor == nil {
+				continue
+			}
+			foreignTableName := field.ForeignKeyDescriptor.ForeignTableName
+			if !knownTables[foreignTableName] && !processedTables[foreignTableName] {
+				newlyReferencedTables = append(newlyReferencedTables, foreignTableName)
+				knownTables[foreignTableName] = true
+			}
+		}
+		tableDescriptorMap[table] = fields
+	}
+	return tableDescriptorMap, newlyReferencedTables, nil
+}
+
+func getInsertionOrder(tablesToFieldsMap map[string][]FieldDescriptor) ([]string, error) {
+	var tablesVisitOrder []string
+	tablesVisited := make(map[string]bool)
```

Comment:
This can be more efficient if you're using `map[string]struct{}` instead of `map[string]bool`.